remote-api/src/main/java/org/alfresco/rest/api/impl/rules/RuleSetsImpl.java

```diff
@@ -59,10 +59,12 @@ public CollectionWithPagingInfo<RuleSet> getRuleSets(String folderNodeId, List<S
     {
         NodeRef folderNode = validator.validateFolderNode(folderNodeId, false);
 
-        NodeRef ruleSetNode = ruleService.getRuleSetNode(folderNode);
-        List<RuleSet> ruleSets = Optional.ofNullable(ruleSetNode)
-                                         .map(nodeRef -> ruleSetLoader.loadRuleSet(nodeRef, folderNode, includes))
-                                                                      .stream().collect(toList());
+        List<RuleSet> ruleSets = ruleService.getNodesSupplyingRuleSets(folderNode)
+                                            .stream()
+                                            .map(ruleService::getRuleSetNode)
+                                            .filter(Objects::nonNull)
+                                            .map(nodeRef -> ruleSetLoader.loadRuleSet(nodeRef, folderNode, includes))
+                                            .collect(toList());

```

Comment:
Is there a well-defined order to the rule sets (given that in theory they can be paged) ?

Actually, maybe my question is answered further below, ie. getNodesSupplyingRuleSets(folderNode:

> The order of nodes returned by this methods has to match the order used by {@link #getInheritedRules}.

Reply:
I have reproduced the existing rule order from the v0 API (which is the order in which the rules will be executed).

The order is top-down for a simple hierarchy. When we start adding secondary parents in then the order is not clear to me - I assume it appears consistent because the db decides to return things in a fixed order (it's not based on creation date or anything like that).

I have an additional test that uses secondary parents to demonstrate this issue, which we discussed here: https://alfresco.slack.com/archives/CTWMLBG6N/p1660832592553109

Reply:
Having spent yesterday investigating the test I mentioned then I now feel more confident that it is working as expected. I will push it as a separate PR.